src/js/common/schemaform/SaveFormLink.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import LoginModal from '../components/LoginModal';
+
+// TODO: Come up with a better name than SaveFormLink
+class SaveFormLink extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalOpened: false
+    };
+  }
+
+  openLoginModal = () => {
+    this.setState({ modalOpened: true });
+  }
+
+  closeLoginModal = () => {
+    this.setState({ modalOpened: false });
+  }
+
+  render() {
+    const {
+      saveForm,
+      savedStatus
+    } = this.props;
+    let content = (<div>
+      <a onClick={this.openLoginModal}>Sign in before saving your application</a>
+      <LoginModal
+          key={1}
+          title="Sign in to save your application"
+          onClose={this.closeLoginModal}
+          visible={this.state.modalOpened}/>
+    </div>);
+
+    if (this.props.loggedIn) {
+      content = <a onClick={saveForm}>Save and come back later</a>;
+    }
+
+    if (savedStatus === 'no-auth') {
```

Comment:
Yeah, I that was one of those things I meant to do but never got around too. Will do!